main/src/cgeo/geocaching/maps/MapDistanceDrawerCommons.java

```diff
@@ -28,27 +31,34 @@ public MapDistanceDrawerCommons(final View root) {
 
         distance1InfoView.setOnClickListener(v -> swap());
         distance1View.setOnClickListener(v -> swap());
+        distance2InfoView.setOnClickListener(v -> swap());
+        distance2View.setOnClickListener(v -> swap());
+        routeDistanceView.setOnClickListener(v -> swap());
         distanceSupersizeView.setOnClickListener(v -> swap());
     }
 
     public void drawDistance(final boolean showBothDistances, final float distance, final float realDistance) {
-        final boolean bothViewsNeeded = showBothDistances && realDistance != 0.0f && distance != realDistance;
-        final boolean supersize = Settings.getSupersizeDistance();
-        final TextView d = supersize ? distanceSupersizeView : distance1View;
+        bothViewsNeeded = showBothDistances && realDistance != 0.0f && distance != realDistance;
+        final int supersize = Settings.getSupersizeDistance() % (bothViewsNeeded ? 3 : 2);
```

Comment:
I have a bit of a bad feeling with these "magic" supersize numbers (0, 1 and 2). Is it an option to use an enum instead.
Also, the code seems like it would be vulnerable if "bothViewsNeeded" is switched from true to false and vice versa in consective cars. I think this is actually not the case since a previously stored supersize value of 2 would be reduced to 0 in case bothViewsNeeded=false, which as far as I understand the code triggers the "correct" behaviour of not displaying a supersize. But there is some hidden magic in there which is not obvious imho.

Reply:
Thanks for cross-checking. Those "magic numbers" are actually nothing more than a plain counter - first (1), second (2) or no (0) distance window being super-sized. "Second" only being possible if "bothViews" setting is activated and a calculated distance is available (resulting in `bothViewsNeeded == true`). And, exactly, fallback is intended as you've described. - Ok for you, or shall I change this to either using named constants or an enum class?

Reply:
ok for me, leave as is